### ISSUE TITLE
Executing POST request instead of GET

### ISSUE THREAD

**Issue by AnatolyRugalev** [yiisoft/yii2-httpclient#27]
I'm using this extension to talk with Gitlab REST API and faced issue that GET requests are not performing right in some cases.

```php
class Gitlab extends \yii\httpclient\Client
{
    public $privateToken;
    public $projectId;

    public function createRequest()
    {
        $request = parent::createRequest();
        $request->setFormat(self::FORMAT_JSON);
        $request->addHeaders([
            'PRIVATE-TOKEN' => $this->privateToken,
        ]);
        return $request;
    }
}

$response = $gitlab->get("projects/{$gitlab->projectId}/repository/branches/{$prefix}{$issue}")->send();
if ($response->statusCode == 200) {
    // Do something
}
```

httpclient logs that GET request execution is being performed but in fact, a POST request is performed. Debugging led me to `CurlTransport:123`:

```php
switch ($method) {
    case 'GET':
        break;
    case 'POST':
        $curlOptions[CURLOPT_POST] = true;
        break;
    default:
        $curlOptions[CURLOPT_CUSTOMREQUEST] = $method;
}
```

When I deleted `case 'GET'` part, I got an expected result.

Any thoughts about this? Is it neccessary to NOT set `CURLOPT_CUSTOMREQUEST` option in case of GET and POST? Maybe we can set it always to desired method?

**Comment by AnatolyRugalev** [comment]
We can add test cases bases on [httpbin.org](https://httpbin.org/) to verify request verbs and headers

**Comment by klimov-paul** [comment]
Resolved by commit 6d5bb61c10a8a27221cb6e95bc9ea314f6f371d6
